src/content/docs/ru/guides/testing.mdx

```diff
@@ -6,25 +6,257 @@ i18nReady: true
 import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
 
 
-Тестирование поможет вам писать и поддерживать работающий Astro код. Astro поддерживает множество популярных инструментов для юнит, компонентных и end-to-end тестов включая Jest, Mocha, Jasmine, Cypress и Playwright. Вы также можете установить библиотеки тестирования, специфичные для фреймворка, такие как React Testing Library, для тестирования ваших UI компонент.
+Тестирование поможет вам писать и поддерживать работающий Astro код. Astro поддерживает множество популярных инструментов для юнит, компонентных и end-to-end тестов включая Jest, Mocha, Jasmine, Cypress и Playwright. Вы также можете установить библиотеки тестирования, специфичные для фреймворка, такие как React Testing Library, для тестирования ваших UI-компонентов.
 
 Фреймворки тестирования позволяют вам указывать **утверждения** или **ожидания** того, как ваш код должен вести себя в конкретных ситуациях, а затем сравнивать их с фактическим поведением вашего кода. 
 
 ## Vitest
 
 Нативный Vite фреймворк для юнит тестирования с поддержкой ESM, TypeScript и JSX на базе esbuild.
 
+Используйте хелпер Astro `getViteConfig()` в вашем конфигурационном файле [`vitest.config.ts`](https://vitest.dev/config/), чтобы настроить Vitest с учётом настроек вашего проекта Astro:
+
+```js
+// vitest.config.ts
+import { getViteConfig } from 'astro/config';
+
+export default getViteConfig({
+  test: {
+    // Варианты конфигурации Vitest
+  },
+});
+```
+
 Посмотрите [стартовый шаблон Astro + Vitest](https://github.com/withastro/astro/tree/latest/examples/with-vitest) в GitHub.
 
+## Cypress
+
+Cypress — это инструмент для фронтального тестирования, созданный для современного веба. Cypress позволяет писать сквозные тесты для вашего сайта Astro.
+
+### Установка
+
+Вы можете установить Cypress с помощью выбранного вами менеджера пакетов.
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm install -D cypress
+  ```
+
+  Это позволит установить Cypress локально как зависимость dev для вашего проекта.
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm add cypress --save-dev
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn add cypress --dev
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+### Конфигурация
+
+В корне проекта создайте файл `cypress.config.js` со следующим содержимым:
+
+```js title="cypress.config.js"
+import { defineConfig } from 'cypress'
+
+export default defineConfig({
+  e2e: {
+    supportFile: false
+  }
+})
+```
+
+### Создайте свой первый Cypress тест
+
+1. Выберите страницу для тестирования. В этом примере будет протестирована страница `index.astro`, приведённая ниже.
+
+    ```html title="src/pages/index.astro"
+    ---
+    ---
+    <html lang="en">
```

Comment:
I've noticed that you were translating content within syntax-highlighted code, but it is not consistent. We can either keep the original English content, or translate everything uniformly. I lean towards keeping the original English content in the code.

Reply:
I'll try to translate everywhere, but later. After all, it is preferable for readers to see comments in the language of their choice :)

![sshot-15](https://github.com/withastro/docs/assets/229402/53eab4a2-feac-49f9-9978-5f3c83076a7b)